honeybee_energy/measure.py

```diff
@@ -0,0 +1,428 @@
+                base['resource_data'][rb_file] = self._compress_file(path)
+
+        return base
+
+    def to_osw_dict(self):
+        """Get a Python dictionary that can be written to an OSW JSON.
+
+        Specifcally, this dictionary can be appended to the "steps" key of the
+        OpenStudio Workflow (.osw) JSON dictionary in order to include the measure
+        in the workflow.
+
+        Note that this method does not perform any checks to validate that the
+        Measure has all required values and only arguments with values will be
+        included in the dictionary. Validation should be done seperately with
+        the validate method.
+        """
+        base = {'measure_dir_name': self.folder, 'arguments': {}}
+        for arg in self._arguments:
+            if arg.value is not None:
+                base['arguments'][arg.identifier] = arg.value
+        return base
+
+    def validate(self, raise_exception=True):
+        """Check if all required arguments have values needed for simulation.
+
+        Args:
+            raise_exception: If True, an exception will be raised if there's a
+                required argument and there is no value. Otherwise, False will
+                be returned for this case and True will be returned if all
+                is correct.
+        """
+        for arg in self._arguments:
+            if not arg.validate(raise_exception):
+                return False
+        return True
+
+    @staticmethod
+    def sort_measures(measures):
+        """Sort measures according to the order they will be executed by OpenStudio CLI.
+
+        ModelMeasures will be first, followed by EnergyPlusMeasures, followed by
+        ReportingMeasures.
+        """
+        m_dict = {'ModelMeasure': [], 'EnergyPlusMeasure': [], 'ReportingMeasure': []}
+        for measure in measures:
+            m_dict[measure.type].append(measure)
+        return m_dict['ModelMeasure'] + m_dict['EnergyPlusMeasure'] + \
+            m_dict['ReportingMeasure']
+
+    def _parse_xml_path(self):
+        """Parse measure properties from the measure.xml file."""
+        # create an element tree object
+        tree = ElementTree.parse(self._xml_path)
+        root = tree.getroot()
+
+        # parse the measure properties from the element tree
+        self._identifier = root.find('name').text
+        self._display_name = root.find('display_name').text
+        self._description = root.find('description').text
+        self._measure_type = None
+        for atr in root.find('attributes'):
+            if atr.find('name').text == 'Measure Type':
+                self._type = atr.find('value').text
+
+        # parse the measure arguments
+        self._arguments = []
+        for arg in root.find('arguments'):
+            arg_obj = MeasureArgument(arg)
+            if arg_obj.model_dependent:
+                # TODO: Figure out how to implement model-dependent arguments
```

Comment:
With our current implementation it should be so hard to map the objects from an OpenStudio model to a Honeybee model but I'm looking forward to see what will be your decision on this.

Reply:
I don't follow. In what workflow are you thinking that a measure like that will be used?

Reply:
My understanding is that measures that are model dependent are intended for the case where you already have an OpenStudio model, like when you are in OpenStudio GUI. Not like measures that you would include in a workflow like what we primary do with them.